src/Microsoft.DocAsCode.Common/FileAbstractLayer/LinkFileSystem.cs

```diff
@@ -0,0 +1,277 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Common
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.IO;
+    using System.Linq;
+
+    public class LinkFileSystem
+    {
+
+        #region Consts/Fields
+        private const int MaxRetry = 3;
+        private readonly List<PathMapping> _outputList = new List<PathMapping>();
+        private Dictionary<RelativePath, string> _allInputs;
+        #endregion
+
+        #region Properties
+
+        public ImmutableArray<PathMapping> Mappings { get; }
+
+        public string OutputFolder { get; }
+
+        public bool IsReadOnly => string.IsNullOrEmpty(OutputFolder);
+
+        #endregion
+
+        #region Constructors
+
+        public LinkFileSystem(params PathMapping[] mappings)
+            : this(mappings, null) { }
+
+        public LinkFileSystem(IEnumerable<PathMapping> mappings, string outputFolder = null)
```

Comment:
why?

---
In reply to: [89708316](https://github.com/dotnet/docfx/pull/926#discussion_r89708316) [](ancestors = 89708316)